FlashApp.sh

```diff
@@ -0,0 +1,247 @@
+#!/bin/sh
+
+# Check and if exists, source these files /etc/include.properties and /etc/device.properties
+if [ -f /etc/include.properties ]; then
+    . /etc/include.properties
+fi
+if [ -f /etc/device.properties ]; then
+    . /etc/device.properties
+fi
+
+if [ -z "$PERSISTENT_PATH" ]; then
+    PERSISTENT_PATH="/opt"
+fi
+
+if [ -z "$LOG_PATH" ]; then
+    LOG_PATH="/opt/logs"
+    mkdir -p $LOG_PATH
+fi
+
+FLASHAPPLOGFILE="$LOG_PATH/FlashApp.log"
+EXTBLOCK="$PERSISTENT_PATH/ota/extblock"
+WICIMAGEFILE="$PERSISTENT_PATH/ota/wicimage"
+
+logger() {
+    echo "$(date) FlashApp.sh > $1" | tee -a $FLASHAPPLOGFILE
+}
+
+# Check if all the commands used in this script are available
+commandsRequired="date echo exit ls grep mkdir tar cp rm sync stat losetup mount umount reboot df sed awk md5sum trap"
+for cmd in $commandsRequired; do
+    if ! command -v $cmd > /dev/null; then
+        logger "Required command '$cmd' not found; cannot proceed, exiting."
+        exit 1
+    fi
+done
+
+# This script can be invoked with 2 arguments or 1 argument. Handle both cases.
+# 1. /usr/bin/FlashApp "$DOWNLOAD_LOCATION/$UPGRADE_FILE"
+# 2. /usr/bin/FlashApp "$DOWNLOAD_LOCATION" "$UPGRADE_FILE"
+if [ $# -eq 2 ]; then
+    cloudFWFile="$1/$2"
+    logger "Invoked with two arguments '$1' and '$2'"
+elif [ $# -eq 1 ]; then
+    cloudFWFile="$1"
+    logger "Invoked with single argument '$1'"
+else
+    logger "Invalid number of arguments passed"
+    echo "Usage: $0 <Absolute path to Firmware Image File>"
+    exit 1
+fi
+
+md5sumFile=$(md5sum $cloudFWFile | cut -d' ' -f1)
+logger "Firmware image file received: '$cloudFWFile' with md5sum '$md5sumFile'"
+
+# RPI OTA image is a compressed file; extract it if so.
+if [ $(ls $cloudFWFile | grep -c "tar.gz") -eq 1 ]; then
```

Comment:
What if its compressed by any other format other than tar.gz ? such as tar.bz2

Reply:
The file is being created as a tar.gz archive by the script do_create_ota_image_wic at [this step](https://github.com/rdkcentral/meta-vendor-raspberrypi-release/blob/b8430bd92695df1d88065bc73bfceee7c4afb3bd/conf/machine/include/rpi-ota-image.inc#L75). If any changes override this behavior, the OTA upgrade feature's sanity testing will fail. We can update the logic accordingly when that happens.